lib/src/main/kotlin/com/swmansion/starknet/data/types/Responses.kt

```diff
@@ -167,7 +168,7 @@ data class BlockWithTransactionsResponse(
 
 @Serializable
 data class PendingBlockWithTransactionsResponse(
-    // Not in RPC schema
+    // Not in RPC schema, but can be returned by nodes
```

Comment:
how is that? :D

Reply:
Just an observation, it was noticed when using our node.

Reply:
so let's clarify the comment "not in schema, but can work sometimes" actually says very little

Reply:
Do you have something specific to add here in mind? I added this clarification so we know this is not gateway specific field and that it's probably shouldn't be removed despite not being present in RPC.

Reply:
The first part of Tomek's comment should be enough context for any future dev

Reply:
I'm not sure that part actually gives any info, because it refers to the endpoint `getBlockWithTxs` in general. It's true that some nodes do not even support this endpoint for pending block, but the response for pending block is present in the spec.

So I don't see how that info is relevant in that particular case, because this dataclass is already contains `pending` in its name. 

Maybe there was some confusion, because my comment only relates to a field, not entire dataclass, dataclass is very much present in the spec. 

@tomek0123456789 can you please confirm or clarify this? 😅

Reply:
Ah I see - so the method is in spec (and with this body) expect for status field which is added by us (and that can sometimes be returned by some nodes), is that correct?

If it is correct, then:
1. we set the status so I don't think it can/should be overwritten by the node's response
2. I'd rather treat it as some "value added" by us, besides as you mentioned `pending` is already in the name and this status should never shange
3. we can mark it via comment that it is not in spec, but because of 1. I don't think "but can be returned by nodes" is relevant in our case (as the status cannot be overriden) and we should probably ignore such field in node's response (as it is not in spec - and perhaps the spec is what we want to align to, not some random node implementation)

what do you guys think?

Reply:
Oh, a misunderstanding, I thought this comment was about the class itself, not the field 😅, sorry for that.

Reply:
1. Yes, the value is set no matter what node returns.
2. I kind of agree, but I wanted to specify that aside from being "value added by us", this is something that actually reflects behaviour of some RPC nodes, which means it's not a gateway specific thing and not something to get rid of when we start removing other gateway-specific fields that are marked as `Not in RPC schema` as well.
3. I agree that it generally is irrelevant information because we are ignoring that field, but the idea was to specify that our "custom value" is something that can potentially "conflict" with what node returns. 

If you think we should have cleaner comments I don't mind reverting this one.

Reply:
Sure, I understand. Whatever suits you - I would probably remove this comment, but won't oppose if you decided to leave it :+1: